cmd/deploy.go

```diff
@@ -311,33 +311,7 @@ func autofillApimodel(dc *deployCmd) error {
 
 	k8sConfig := dc.containerService.Properties.OrchestratorProfile.KubernetesConfig
 
-	useManagedIdentity := k8sConfig != nil &&
-		k8sConfig.UseManagedIdentity
-
-	if dc.containerService.Properties.MasterProfile.IsVirtualMachineScaleSets() {
-		k8sConfig.UserAssignedID = acsengine.DefaultUserAssignedID
-	}
-	userAssignedID := k8sConfig != nil &&
-		k8sConfig.UseManagedIdentity &&
-		k8sConfig.UserAssignedID != ""
-
-	// Note: User assigned identity can be assigned from the ARM template, but the role assigment following that will
-	// fail due to a bug with the service. This code is added to wait till the newly created AAD identity is properly
-	// propogated.
-	if userAssignedID {
-		userID, err := dc.client.CreateUserAssignedID(dc.location, dc.resourceGroup, k8sConfig.UserAssignedID)
-		if err != nil {
-			return nil
-		}
-		// Fill up the client id for creating azure.json
-		k8sConfig.UserAssignedClientID = (*userID.ClientID).String()
-		err = dc.client.CreateRoleAssignmentSimple(ctx, dc.resourceGroup, (*userID.PrincipalID).String())
-		if err != nil {
-			return errors.Wrap(err, "apimodel: could not create role assignment for user assigned id ")
-		}
-		//TODO: Support e2e return fake user id.
-		return nil
-	}
+	useManagedIdentity := k8sConfig != nil && k8sConfig.UseManagedIdentity
```

Comment:
There seems to be redundancy with https://github.com/Azure/acs-engine/blob/f351426745bfcebce79e9d4a7834cc37d6541fe7/pkg/acsengine/template_generator.go#L348

We could create a private method and reuse that in the two aforesaid areas. Let's also add a UT for it :)

Reply:
This is just a local variable which was being used, not planning to dip into the global pool of methods for this.

Reply:
Got it